classes/gateways/class.pmprogateway_stripe.php

```diff
@@ -1517,126 +1517,185 @@ function charge( &$order ) {
+				try {
+					$invoice = Stripe_Invoice::retrieve( $order->payment_transaction_id );
+				} catch ( \Throwable $e ) {
+					// Assume no customer found.
+				} catch ( \Exception $e ) {
+					// Assume no customer found.
+				}
+				if ( ! empty( $invoice ) && ! empty( $invoice->customer ) ) {
+					$customer_id = $invoice->customer;
+				}
 			}
 
-			if ( ! empty( $charge ) && ! empty( $charge->customer ) ) {
-				$customer_id = $charge->customer;
+			if ( ! empty( $customer_id ) ) {
+				update_user_meta( $user_id, "pmpro_stripe_customerid", $customer_id );
 			}
 		}
 
-		if ( empty( $customer_id ) && ! empty( $order->payment_transaction_id ) && strpos( $order->payment_transaction_id, "in_" ) !== false ) {
-			// Get the Customer ID from their invoice.
-			try {
-				$invoice = Stripe_Invoice::retrieve( $order->payment_transaction_id );
-			} catch ( \Throwable $e ) {
-				// translators: %s: The error message.
-				$order->error = sprintf( __( 'Error: %s', 'paid-memberships-pro' ), $e->getMessage() );
+		return empty( $customer_id ) ? null : $this->get_customer( $customer_id );
+	}
 
-				return false;
-			} catch ( \Exception $e ) {
-				$order->error = sprintf( __( 'Error: %s', 'paid-memberships-pro' ), $e->getMessage() );
+	/**
+	 * Update a customer in Stripe.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $customer_id to update.
+	 * @param array  $args to update with.
+	 * @return Stripe_Customer|string error message.
+	 */
+	private function update_customer( $customer_id, $args ) {
+		try {
+			$customer = Stripe_Customer::update( $customer_id, $args );
+		} catch ( \Stripe\Error $e ) {
+			return $e->getMessage();
+		} catch ( \Throwable $e ) {
+			return $e->getMessage();
+		} catch ( \Exception $e ) {
+			return $e->getMessage();
+		}
+		return $customer;
+	}
 
-				return false;
-			}
+	/**
+	 * Create a new customer in Stripe.
+	 *
+	 * @since TBD
+	 *
+	 * @param array  $args to update with.
+	 * @return Stripe_Customer|string error message.
+	 */
+	private function create_customer( $args ) {
+		try {
+			$customer = Stripe_Customer::create( $args );
+		} catch ( \Stripe\Error $e ) {
+			return $e->getMessage();
+		} catch ( \Throwable $e ) {
+			return $e->getMessage();
+		} catch ( \Exception $e ) {
+			return $e->getMessage();
+		}
+		return $customer;
+	}
 
-			if ( ! empty( $invoice ) && ! empty( $invoice->customer ) ) {
-				$customer_id = $invoice->customer;
-			}
-		}		
+	/**
+	 * Create/Update Stripe customer from MemberOrder.
+	 *
+	 * Falls back on information in User object if insufficient
+	 * information in MemberOrder.
+	 *
+	 * Should only be called when checkout is being proceesed. Otherwise,
+	 * use update_customer_from_user() method.
+	 *
+	 * @since TBD
+	 *
+	 * @param MemberOrder $order to create/update Stripe customer for.
+	 * @return Stripe_Customer|false
+	 */
+	private function update_customer_from_order( $order ) {
```

Comment:
Let's call this update_customer_at_checkout since that's how it's used.
```suggestion
	private function update_customer_at_checkout( $order ) {
```